src/request/export/ExportWords.js

```diff
@@ -21,104 +22,123 @@ class ExportWords extends Nimiq.Observable {
+            event.preventDefault();
+            this._resolve({ success: true });
+        });
+        this._wordsPasswordBox.on(PassphraseBox.Events.SUBMIT, this._passphraseSubmitted.bind(this));
+        $recoveryWordsContinue.addEventListener('click', () => {
+            this._validateWords.reset();
+            window.location.hash = ExportWords.Pages.VALIDATE_WORDS;
+        });
+        this._validateWords.on(ValidateWords.Events.VALIDATED, () => this._resolve({ success: true }));
+        this._validateWords.on(ValidateWords.Events.SKIP, () => this._resolve({ success: true }));
     }
 
     run() {
-        this._privacyWarningPassphraseBox.reset();
-        window.location.hash = ExportWords.Pages.PRIVACY;
+        this._wordsPasswordBox.reset();
+        window.location.hash = ExportWords.Pages.RECOVERY_WORDS_INTRO;
         if (TopLevelApi.getDocumentWidth() > Constants.MIN_WIDTH_FOR_AUTOFOCUS) {
-            this._privacyWarningPassphraseBox.focus();
+            this._wordsPasswordBox.focus();
         }
     }
 
     /**
-     * @param {string} phrase
+     * @param {string} [password]
      */
-    async _passphraseSubmitted(phrase) {
+    async _passphraseSubmitted(password) {
         TopLevelApi.setLoading(true);
-        const passphraseBuffer = phrase ? Utf8Tools.stringToUtf8ByteArray(phrase) : undefined;
+
         /** @type {Key?} */
         let key = null;
-        try {
-            key = await KeyStore.instance.get(this._request.keyInfo.id, passphraseBuffer);
-        } catch (e) {
-            if (e.message === 'Invalid key') {
-                TopLevelApi.setLoading(false);
-                this._privacyWarningPassphraseBox.onPassphraseIncorrect();
+
+        if (this._key) {
+            key = this._key;
+            password = this._password;
+        } else {
+            const passwordBuffer = password ? Utf8Tools.stringToUtf8ByteArray(password) : undefined;
+
+            try {
+                key = await KeyStore.instance.get(this._request.keyInfo.id, passwordBuffer);
+            } catch (e) {
+                if (e.message === 'Invalid key') {
+                    this._wordsPasswordBox.onPassphraseIncorrect();
+                    TopLevelApi.setLoading(false);
+                    return;
+                }
+                this._reject(new Errors.CoreError(e));
                 return;
             }
-            this._reject(new Errors.CoreError(e));
-            return;
-        }
-        if (!key) {
-            this._reject(new Errors.KeyNotFoundError());
-            return;
+
+            if (!key) {
+                this._reject(new Errors.KeyNotFoundError());
+                return;
+            }
+            this.fire(ExportWords.Events.KEY_CHANGED, key, password);
         }
 
-        this.setKey(key);
-        this.fire(ExportWords.Events.KEY_CHANGED, {
-            key,
-            isProtected: this._request.keyInfo.encrypted,
-        });
+        this.setKey(key, password);
```

Comment:
For the `if (this._key)` case, this is redundant, isn't it?

Reply:
It is. Moved it into the else block of said conditional.